qemu/tests/steal_time.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import time
+import re
+
+from virttest import error_context
+from virttest import utils_test
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Steal time test:
+    1) Boot two guests and bind to one same cpu
+    2) Run stress inside both guests
+    3) Check steal time in top inside guests
+    4) Check steal time in /proc/stat in guest
+    5) Repeat step 4) after 60s, compare the steal time changed in two guests
+
+    :params test: QEMU test object.
+    :params params: Dictionary with the test parameters.
+    :params env: Dictionary with test environment.
+    """
+    def get_stat_val():
+        """
+        Get steal time value in /proc/stat
+        """
+        stat_val = []
+        for session in sessions:
+            val = session.cmd_output(stat_cmd).split()[8]
+            stat_val.append(int(val))
+        return stat_val
+
+    stress_args = params["stress_args"]
+    stress_tests = []
+    sessions = []
+    vms = env.get_all_vms()
+
+    error_context.context("Run stress in both guests", logging.info)
+    for vm in vms:
+        session = vm.wait_for_login()
+        sessions.append(session)
+        stress_test = utils_test.VMStress(vm, "stress",
+                                          params, stress_args=stress_args)
+        stress_test.load_stress_tool()
+        stress_tests.append(stress_test)
+
+    time.sleep(10)
+    top_cmd = params["top_cmd"]
+    stat_cmd = params["stat_cmd"]
+
+    try:
+        error_context.context("Check steal time in guests", logging.info)
+        for session in sessions:
+            output = session.cmd_output(top_cmd)
+            top_st = re.findall(r",\s*(\d+.?\d+)\s*st", output)[0]
+            if abs(float(top_st) - 50) > 10:
+                test.fail("Guest steal time is not around 50")
+
+        stat_val_pre = get_stat_val()
+        logging.info("Steal time value in /proc/stat is %s", stat_val_pre)
+        time.sleep(60)
+        stat_val_post = get_stat_val()
+        logging.info("After 60s, steal time value in /proc/stat is %s",
+                     stat_val_post)
+
+        delta = list(map(lambda x, y: y - x, stat_val_pre, stat_val_post))
+        if abs(delta[0] - delta[1]) > delta[1]*0.1:
```

Comment:
I have a bit concern about the threshold `delta[1]*0.1` but no more good idea now. how about use the average `(delta[0] + delta[1]/2)*0.1` ?

Reply:
Actually the difference between delta[0] and delta[1] is very small, much smaller than 10% of both. So I think it's fine. Thanks.

Reply:
'difference is small' is the normal case i think.  I just thought use a value that may be wrong as threshold is not very suitable, so suggest average in the pre comment.
Does fixed threshold work with this situation ?